Rephrase the overview (#478)
tlswg/draft-ietf-tls-esni: draft-ietf-tls-esni.md

## Encrypted ClientHello (ECH)

A client-facing server enables ECH by publishing an ECH configuration, which
is an encryption public key and associated metadata. The server must publish

> this for all the domains it serves directly or indirectly (via Split Mode).

Reviewer: ```suggestion
this for all the domains it serves directly or indirectly via Shared or Split Mode, respectively.
```

Author: Oh, I got this from the earlier version of the text. I wonder whether both OR clauses are even necessary. WDYT about:

> for all the domains it serves via Shared or Split Mode.

Author: Applied that version and rebased.

Reviewer: Lovely :-) thanks!

This document defines the ECH configuration's format, but delegates DNS
publication details to {{!HTTPS-RR=I-D.ietf-dnsop-svcb-https}}. Other delivery
mechanisms are also possible. For example, the client may have the ECH
configuration preconfigured.

When a client wants to establish a TLS session with some backend server, it

> constructs a secret ClientHello, referred to as the ClientHelloInner.

Reviewer: ```suggestion
constructs a private ClientHello, referred to as the ClientHelloInner.
```

The client then constructs a public ClientHello, referred to as the
ClientHelloOuter. The ClientHelloOuter contains innocuous values for
sensitive extensions and an "encrypted_client_hello" extension
({{encrypted-client-hello}}), which carries the encrypted ClientHelloInner.
Finally, the client sends ClientHelloOuter to the server.

The server takes one of the following actions:

1. If it does not support ECH or cannot decrypt the extension, it completes
   the handshake with ClientHelloOuter. This is referred to as rejecting ECH.
1. If it successfully decrypts the extension, it forwards the ClientHelloInner
   to the backend server, which completes the handshake. This is referred to
   as accepting ECH.

Upon receiving the server's response, the client determines whether or not ECH
was accepted ({{handle-server-response}}) and proceeds with the handshake

> accordingly. When ECH is rejected, the connection is not directly usable for
> the client, but allows the client to retry with up-to-date configuration
> ({{rejected-ech}}).

Reviewer: ```suggestion
accordingly. When ECH is rejected, the resulting connection is not usable by
the client for application data. Instead, ECH rejection allows the client to
retry with up-to-date configuration ({{rejected-ech}}).
```
